test/lib_safemath64.js

```diff
@@ -0,0 +1,54 @@
+const { assertRevert } = require('./helpers/assertThrow')
+
+contract('SafeMath64 lib test', accounts => {
+  let safeMath64Mock
+
+  before(async () => {
+    safeMath64Mock = await artifacts.require('SafeMath64Mock').new()
+  })
+
+  // multiplication
+  it('multiplies', async () => {
+    const a = 1234
+    const b = 32746
+    assert.equal((await safeMath64Mock.mulExt(a, b)).toString(), a * b, "Values should match")
+  })
+
+  it('fails if multplication overflows', async () => {
+    const a = new web3.BigNumber(2).pow(63)
+    const b = 2
+    return assertRevert(async () => {
+      await safeMath64Mock.mulExt(a, b)
+    })
+  })
+
+  // subtraction
+  it('subtract', async () => {
+    const a = 1234
+    const b = 327
+    assert.equal((await safeMath64Mock.subExt(a, b)).toString(), a - b, "Values should match")
+  })
+
+  it('fails if subtraction underflows', async () => {
+    const a = 123
+    const b = 124
+    return assertRevert(async () => {
+      await safeMath64Mock.subExt(a, b)
+    })
+  })
+
+  // addition
+  it('adds', async () => {
+    const a = 1234
+    const b = 327
+    assert.equal((await safeMath64Mock.addExt(a, b)).toString(), a + b, "Values should match")
+  })
+
+  it('fails if addition overflows', async () => {
+    const a = new web3.BigNumber(2).pow(63)
+    const b = new web3.BigNumber(2).pow(63)
+    return assertRevert(async () => {
+      await safeMath64Mock.addExt(a, b)
+    })
+  })
+})
```

Comment:
I guess we might as well add a division case?

Reply:
I didn't add it because I actually I wanted to discuss about it with you: shouldn't we remove division instead? I don't think we are using it, and Solidity is supposed to throw when dividing by zero.

Reply:
Solidity automatically `asserts`, but we should `revert` since all the other operations are reverting.

Reply:
I did a quick test on Remix and gas execution cost went from `359` to `539` from just dividing to using that SafeMathlib function. I don't think it's worth it. Dividing by zero should be the exception so we would be losing gas in most of the cases.

Reply:
Yeah there's definitely some overhead, but I'd strongly suggest keeping it to simplify the mental model when using SafeMath. You know you're protected by a revert when you use it, in the worst case scenario, rather than an assert that will eat up all the gas.